### PR TITLE
updates avo sts polcies to allow for creating addl hosted zones

### PR DESCRIPTION
**Pull request by tonytheleg**
### What type of PR is this?
_(bug/feature/cleanup/documentation)_
feature

### What this PR does / why we need it?
[OSD-13509](https://issues.redhat.com/browse/OSD-13509) adds a feature to AVO that allows configuring hosted zones for MUO access to OCM in FedRAMP that is unreachable without it due to Privatelink. These changes require some extra Route53 permissions in the sts policy

### Which Jira/Github issue(s) this PR fixes?

_Fixes #_ [OSD-13509](https://issues.redhat.com//browse/OSD-13509)

### Special notes for your reviewer:
Changes are being made to 4.10 and 4.11 policies as FedRAMP is still running 4.10 clusters.

### Pre-checks (if applicable):
- [x] Tested latest changes against a cluster
- [x] Included documentation changes with PR
- [x] If this is a new object that is not intended for the FedRAMP environment (if unsure, please reach out to team FedRAMP), please exclude it with:

    ```yaml
    matchExpressions:
    - key: api.openshift.com/fedramp
      operator: NotIn
      values: ["true"]
    ```

### CODE REVIEW

File: resources/sts/4.10/openshift_aws_vpce_operator_avo_aws_creds_policy.json [hunk 1]
```diff
@@ -14,13 +14,17 @@
...
         "ec2:CreateVpcEndpoint",
         "ec2:DeleteVpcEndpoints",
         "ec2:DescribeVpcEndpoints",
         "ec2:ModifyVpcEndpoint",
         "route53:ChangeResourceRecordSets",
         "route53:ListHostedZonesByName",
-        "route53:ListResourceRecordSets"
+        "route53:ListResourceRecordSets",
+        "route53:ListTagsForResource",
+        "route53:CreateHostedZone",
```
Comment (jharrington22) [inline resources/sts/4.10/openshift_aws_vpce_operator_avo_aws_creds_policy.json:26]: @tonytheleg do we need `route53:CreateHostedZone` ? It looks like we [create](https://github.com/openshift/aws-vpce-operator/blob/b1a65a2c26d34e77ae3f30279aa8bad7db4170c4/controllers/vpcendpoint/validation.go#L279) a hosted zone if its not found is that required? What is AddtlHostedZoneName?

cc/ @dustman9000

Comment (tonytheleg) [reply resources/sts/4.10/openshift_aws_vpce_operator_avo_aws_creds_policy.json:26]: The `AddtlHostedZoneName` is new feature add to address [OSD-13509](https://issues.redhat.com/browse/OSD-13509). Details are in that card but essentially we need to be able to create HostedZones for domains for endpoints that wont exist in a customer account in order to properly route traffic to endpoints like OCM in FedRAMP as none of the endpoints are publically available. [OSD-13510](https://issues.redhat.com/browse/OSD-13510) follows up on it to be able to create CNAMES in the additional hosted zone to resolve addresses in the added zone to the correct VPC endpoints,

**Comment by tonytheleg** [comment]
@jharrington22 @mjlshen trying to bump this PR back to life. I know there was conversation about limiting the scope of these STS permissions [HERE](https://github.com/openshift/aws-vpce-operator/pull/132) but it sounds like the impact of those changes may cause issues when customers remove clusters. 

There is an extra layer of time sensitivity here with some patching requirements  for AVO currently needing attention for FedRAMP compliance and we  are unable to promote without breaking the operator. 

Changes added are to reflect the current [requirements](https://github.com/openshift/aws-vpce-operator/blob/main/README.md#requirements) for v1alpha2 for current feature capabilities.

**Comment by tonytheleg** [comment]
@fahlmant updated for the Route53 conditions requested and validated all functionality still works.

**Review by fahlmant — APPROVED** [review]
/label tide/merge-method-squash
/approve
/lgtm

**Pull request merged**